test/unit/test_junit_xml_export.gd

```diff
@@ -29,15 +37,34 @@ func run_scripts(g, one_or_more):
 # Very simple xml validator.  Matches closing tags to opening tags as they
 # are encountered and any validation provided by XMLParser (which is very
 # little).  Does not catch malformed attributes among other things probably.
-func assert_is_valid_xml(s):
+# Additionally checks for valid tags as defined by RESULT_XML_VALID_TAGS
+func assert_is_valid_xml(xml : String)->void:
 	var tags = []
-	var pba = s.to_utf8_buffer()
+	var pba = xml.to_utf8_buffer()
 	var parser = XMLParser.new()
 	var result = parser.open_buffer(pba)
-
+	
 	while(result == OK):
 		if(parser.get_node_type() == parser.NODE_ELEMENT):
-			tags.push_back(parser.get_node_name())
+			var tag_name := parser.get_node_name()
+			tags.push_back(tag_name)
+			
+			if (tag_name in RESULT_XML_VALID_TAGS):
+				# check for required attributes
+				var required_attributes : Array = RESULT_XML_VALID_TAGS[tag_name].duplicate()
+				var missing_attributes := required_attributes.filter(func(attribute): return !parser.has_attribute(attribute))
+				assert_eq(missing_attributes, [], "Required attribute(s) missing.")
+				
+				# check for unexpected attributes
+				var unexpected_attributes : Array[String] = []
+				for attribute_index : int in parser.get_attribute_count():
+					var attribute_name := parser.get_attribute_name(attribute_index)
+					if not attribute_name in RESULT_XML_VALID_TAGS[tag_name]:
+						unexpected_attributes.push_back(attribute_name)
+				assert_eq(unexpected_attributes, [], "Unexpected attribute(s).")
```

Comment:
Change this to the following so the failure message has more details:
```
assert_eq(unexpected_attributes, [], str(tag_name, " Unexpected attribute(s) ", unexpected_attributes))
```

Reply:
Done!